packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/boolean_state_var.test.rs

```diff
@@ -0,0 +1,423 @@
+    let (child_dependency, child_var) = create_state_var_dependency(true, false);
+
+    let dependencies_created_for_data_queries =
+        vec![DependenciesCreatedForDataQuery(vec![child_dependency])];
+
+    state_var.save_dependencies(&dependencies_created_for_data_queries);
+
+    ////////////////////////////////////////////////////////////////
+    // Step 3: check if get the correct calculated value
+    ////////////////////////////////////////////////////////////////
+    state_var.calculate_and_mark_fresh();
+
+    // we expect a value of true and came_from_default to be false,
+    // since that the variable is coming from the child dependency, which we initialized to true.
+    assert_eq!(*state_var.get(), true);
+    assert_eq!(state_var.came_from_default(), false);
+
+    ////////////////////////////////////////////////////////////////
+    // Step 4: invert to make the value false
+    ////////////////////////////////////////////////////////////////
+
+    // on the state variable view, record that we request the value be false
+    state_var_view.queue_update(false);
+
+    let invert_result = state_var.invert(false).unwrap();
+
+    // we should get a request informing core that we need to change the variable
+    assert_eq!(
+        invert_result,
+        vec![DependencyValueUpdateRequest {
+            data_query_idx: 0,
+            dependency_idx: 0
+        }]
+    );
+    // the child variable has recorded that it has been requested to be false
+    assert_eq!(*child_var.get_requested_value(), false);
+
+    ////////////////////////////////////////////////////////////////
+    // Step 5: make the changes to actually make the value false
+    ////////////////////////////////////////////////////////////////
+
+    child_var.set_value(false);
+
+    state_var.calculate_and_mark_fresh();
+
+    // now the value should be false
+    assert_eq!(*state_var.get(), false);
+}
+
+/// Testing the case of boolean state variable requesting children
+/// and receiving a single string child dependency.
+/// For example, if we have a `<boolean>true</boolean>`,
+/// its `value` state variable would follow this pattern.
+#[test]
+fn boolean_state_var_with_string_child() {
+    // create a boolean state variable requesting children
+    let mut state_var = BooleanStateVar::new_from_children(false).into_state_var();
+    let mut state_var_view = state_var.create_new_read_only_view();
+
+    //////////////////////////////////////////////////
+    // Step 1: check that get the correct data queries
+    //////////////////////////////////////////////////
+    let queries = state_var.return_data_queries();
+
+    // should return a query for text or boolean children
+    assert_eq!(
+        queries,
+        vec![DataQuery::Child {
+            match_profiles: vec![ComponentProfile::Text, ComponentProfile::Boolean],
+            exclude_if_prefer_profiles: vec![],
+            always_return_value: true,
+        },]
+    );
```

Comment:
This is the same as above, right? We could split this off into its own test, and then only do it once.

Reply:
Yes, I suppose that would make sense. I'm still thinking in terms of integration or end-to-end tests where a test does a whole sequence of things.